packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
+        } else {
+            this.onCustomPolicyCheckResponse.fire(['Reference policy document is missing.', 'red'])
+            return
+        }
+        switch (documentType) {
+            case PolicyChecksDocumentType.TERRAFORM_PLAN: {
+                const tfCommand = `tf-policy-validator check-access-not-granted --template-path ${document} --region ${this.region} --config ${defaultTerraformConfigPath} --actions ${referenceDocument}`
+                this.executeCustomPolicyChecksCommand(tfCommand)
+                return
+            }
+            case PolicyChecksDocumentType.CLOUDFORMATION: {
+                const cfnCommand =
+                    cfnParameterPath === ''
+                        ? `cfn-policy-validator check-access-not-granted --template-path ${document} --region ${this.region} --actions ${referenceDocument}`
+                        : `cfn-policy-validator check-access-not-granted --template-path ${document} --region ${this.region} --actions ${referenceDocument} --template-copnfiguration-file ${cfnParameterPath}`
+                this.executeCustomPolicyChecksCommand(cfnCommand)
+                return
+            }
+        }
+    }
+
+    public executeValidatePolicyCommand(command: string) {
+        try {
+            this.handleValidatePolicyCliResponse(execSync(command).toString())
+        } catch (err: any) {
+            this.onValidatePolicyResponse.fire([getCliErrorMessage(err.message), 'red'])
+        }
+    }
+
+    public handleValidatePolicyCliResponse(response: string) {
+        const diagnostics: vscode.Diagnostic[] = []
+        const jsonOutput = JSON.parse(response)
+        if (jsonOutput.BlockingFindings.length === 0 && jsonOutput.NonBlockingFindings.length === 0) {
+            this.onValidatePolicyResponse.fire([
+                'Policy checks did not discover any problems with your policy.',
+                'green',
+            ])
+        } else {
+            jsonOutput.BlockingFindings.forEach((finding: any) => {
+                diagnostics.push(
+                    new vscode.Diagnostic(
+                        new vscode.Range(
+                            finding.details.locations[0].span.start.line,
+                            finding.details.locations[0].span.start.offset,
+                            finding.details.locations[0].span.end.line,
+                            finding.details.locations[0].span.end.offset
+                        ),
+                        `${finding.findingType}: ${finding.code} - ${finding.details.findingDetails} Learn more: ${finding.details.learnMoreLink}`,
+                        vscode.DiagnosticSeverity.Error
+                    )
+                )
+                validatePolicyDiagnosticCollection.set(IamPolicyChecksWebview.editedDocumentUri, diagnostics)
+            })
+            jsonOutput.NonBlockingFindings.forEach((finding: any) => {
+                diagnostics.push(
+                    new vscode.Diagnostic(
```

Comment:
i see this logic being repeated at multiple places, should it be a helper method instead?